src/include/sof/ipc/msg.h

```diff
@@ -71,16 +71,8 @@ static inline void ipc_msg_free(struct ipc_msg *msg)
 	if (!msg)
 		return;
 
-	struct ipc *ipc = ipc_get();
-	uint32_t flags;
-
-	spin_lock_irq(&ipc->lock, flags);
-
-	list_item_del(&msg->list);
 	rfree(msg->tx_data);
 	rfree(msg);
-
-	spin_unlock_irq(&ipc->lock, flags);
```

Comment:
This doesn't seem right. If the IPC message can be on the list, it has to be removed from it and that has to be protected, because `ipc` is a global object - common for all cores.

Reply:
right ok. fixed this now.

Reply:
i've dropped this one for now. I think I need to do more testing with this one.